---
title: "TRAIL IZI Melanoma"
author: "A. Ginolhac"
date: "2021-06-21"
output: html_document
---

```{r setup, include = FALSE}
knitr::opts_chunk$set(echo = TRUE)
library(tidyverse)
library(vroom)
library(ggrepel)
library(DESeq2)         # Bioconductor
library(ComplexHeatmap) # Bioconductor
source("R/utils.R")
```


## Summary statistics

Supplementary table XXX: Summary statistics of the sequencing reads and downstream analyses: trimming, mapping and feature assignments

Using tables from [`multiqc`](https://multiqc.info/) also generated by the `snakemake` pipeline.

```{r}
samples <- vroom("config/samples.tsv")
```
- Adapter Removal

```{r paged.print=TRUE}
ar <- vroom("multiqc_report_data/multiqc_adapter_removal.txt") %>% 
  select(Sample, reads_total, reads_retained = retained) %>% 
  mutate(sample = str_extract(Sample, "^[^\\.]+")) %>% 
  left_join(samples) %>% 
  select(-Sample)
```

- STAR

```{r}
star <- vroom("multiqc_report_data/multiqc_star.txt") %>% 
  select(Sample, reads_retained = total_reads, avg_input_read_length, uniquely_mapped, uniquely_mapped_percent) %>% 
  mutate(sample = Sample) %>% 
  left_join(samples) %>% 
  right_join(ar)
```

- featureCounts

```{r}
vroom("multiqc_report_data/multiqc_featureCounts.txt") %>% 
  select(Sample, Assigned) %>% 
  mutate(Sample = str_extract(Sample, "\\d+-\\d")) %>% 
  right_join(star) %>% 
  select(Sample, condition, reads_total, reads_retained, avg_input_read_length,
         uniquely_mapped, uniquely_mapped_percent, assigned = Assigned) %>% 
  mutate(assigned_percent = assigned / uniquely_mapped * 100)
```


Summary statistics of the sequencing reads and downstream analyses: trimming, mapping and feature assignments

**reads_total**: number of raw single-end reads produced by the NextSeq500 machine
**reads_retained**: number of single-end reads after trimming of the raw reads (i.e trimmed reads)
**avg_input_read_length**: average length of trimmed single-end reads
**uniquely_mapped**: number of trimmed reads mapped uniquely by the STAR mapper
**uniquely_mapped_percent**: % of reads mapped uniquely by the STAR mapper from trimmed reads 
**assigned**: number of uniquely mapped reads assign to a gene feature by featureCounts (RSubread)
**assigned_percent**: % of  mapped reads assign to a gene feature from uniquely mapped reads


## Differential gene expression

### PCA

The binary file \`deseq2/all.rds\` from the `Snakemake` pipeline (`DESeq2` init) once loaded is the `dds` object
The binary file \`counts/fc.rds\` from the `Snakemake` pipeline (`featuresCounts` step) once loaded is the `fc` object


```{r}
dds <- read_rds("deseq2/all.rds")
fc <- read_rds("counts/fc.rds")
vst_counts <- vst(dds, blind = TRUE)
pca_df <- plotPCA(vst_counts, intgroup = "condition", returnData = TRUE)
percentVar <- round(100 * attr(pca_df, "percentVar"))
pca_df %>% 
  mutate(IZI = if_else(str_detect(condition, "IZI"), TRUE, FALSE),
         TRAIL = case_when(
           str_detect(condition, "^p") ~ "parental",
           str_detect(condition, "^c") ~ "conditioned",
           TRUE ~ "NA"),
         cell = case_when(
           str_detect(condition,"A375") ~ "A375",
           str_detect(condition, "WM1346") ~ "WM1346",
           TRUE ~ "NA")) -> pca_df_meta
```



### Volcano plots

- To reproduce Fig 4A


```{r}
# normalization and preprocessing
dds_a375 <- subset_dds(dds, pca_df_meta, fc, .cell = "A375")
# 2 contrasts for the volcanoes
res_a375_izic <- sub_results(dds_a375, .coef = "condition_cA375_IZI50_vs_cA375")
res_a375_izip <- sub_results(dds_a375, .coef = "condition_pA375_IZI50_vs_pA375")
# last 2 for the supp tables
res_a375cp <- sub_results(dds_a375, .coef = "condition_cA375_vs_pA375")
res_a375_izicp <- sub_results(dds_a375, .coef = "condition_cA375_IZI50_vs_pA375_IZI50")
# merge all results in one tibble
bind_rows(cIZI50_vs_c = rownames_to_column(as.data.frame(res_a375_izic), var = "gene_id"),
          pIZI50_vs_p = rownames_to_column(as.data.frame(res_a375_izip), var = "gene_id"), 
          .id = "contrast") %>% 
  as_tibble() -> res_tib_a375_izi

two_volcanoes(res_tib_a375_izi, .title = "A375")

# normalization and preprocessing
dds_WM1346 <- subset_dds(dds, pca_df_meta, fc, .cell = "WM1346")
# 2 contrasts for the volcanoes
res_WM1346_izic <- sub_results(dds_WM1346, .coef = "condition_cWM1346_IZI50_vs_cWM1346")
res_WM1346_izip <- sub_results(dds_WM1346, .coef = "condition_pWM1346_IZI50_vs_pWM1346")
# last 2 for the supp tables
res_WM1346cp <- sub_results(dds_WM1346, .coef = "condition_cWM1346_vs_pWM1346")
res_WM1346_izicp <- sub_results(dds_WM1346, .coef = "condition_cWM1346_IZI50_vs_pWM1346_IZI50")
# merge all results in one tibble
bind_rows(cIZI50_vs_c = rownames_to_column(as.data.frame(res_WM1346_izic), var = "gene_id"),
          pIZI50_vs_p = rownames_to_column(as.data.frame(res_WM1346_izip), var = "gene_id"), 
          .id = "contrast") %>% 
  as_tibble() -> res_tib_WM1346_izi

two_volcanoes(res_tib_WM1346_izi, .title = "WM1346") +
  # set same y limits as for the A375
  ylim(c(0, max(-log10(res_tib_a375_izi$padj), na.rm = TRUE)))
```

- Table S2, S4, S5 are obtained for merging the 2 DESeq2 results that contains the gene of interest

```{r paged.print=FALSE}
bind_rows(
  mutate(res_tib_a375_izi, cell = "A375"),
  bind_rows(c_vs_p = rownames_to_column(as.data.frame(res_a375cp), var = "gene_id"),
            cIZI50_vs_pIZI50 = rownames_to_column(as.data.frame(res_a375_izicp), var = "gene_id"), 
            .id = "contrast") %>% mutate(cell = "A375"),
  mutate(res_tib_WM1346_izi, cell = "WM1346"),
  bind_rows(c_vs_p = rownames_to_column(as.data.frame(res_WM1346cp), var = "gene_id"),
            cIZI50_vs_pIZI50 = rownames_to_column(as.data.frame(res_WM1346_izicp), var = "gene_id"), 
            .id = "contrast") %>% mutate(cell = "WM1346")) %>%
  filter(gene_name %in% c("PTK2", "IQGAP1", "ITGAV", "PRKDC", "MET",
                          "BAK", "BAX", "BBC3")) %>%
  select(stimulation = contrast, log2FoldChange, padj, gene_name, cell) %>%
  pivot_wider(id_cols = c(stimulation, gene_name),
              names_from = cell,
              values_from = c(log2FoldChange, padj)) %>%
  arrange(gene_name, stimulation) %>%
  select(stimulation, gene_name, log2FoldChange_A375, padj_A375,
         log2FoldChange_WM1346, padj_WM1346)
```

which gives a table like this (first 10 lines displayed)

```
# A tibble: 28 x 6
   stimulation      gene_name log2FoldChange_A375  padj_A375 log2FoldChange_WM1346 padj_WM1346
   <chr>            <chr>                   <dbl>      <dbl>                 <dbl>       <dbl>
 1 c_vs_p           BAX                    0.331  0.160                    -0.125     0.532   
 2 cIZI50_vs_c      BAX                    0.122  0.377                     0.483     0.000409
 3 cIZI50_vs_pIZI50 BAX                    0.403  0.0566                   -0.0748    0.662   
 4 pIZI50_vs_p      BAX                    0.191  0.361                     0.430     0.000546
 5 c_vs_p           BBC3                   0.332  0.405                     0.0294    0.893   
 6 cIZI50_vs_c      BBC3                   0.0154 0.882                    -0.0745    0.309   
 7 cIZI50_vs_pIZI50 BBC3                   0.385  0.276                    -0.783     0.0663  
 8 pIZI50_vs_p      BBC3                   0.0366 0.908                     0.301     0.329   
 9 c_vs_p           IQGAP1                -0.467  0.00000432                0.106     0.592   
10 cIZI50_vs_c      IQGAP1                 0.0594 0.615                    -0.227     0.0854 
```

### Regularized log counts

- Figs 4C/5A/5B/6D: rlog counts: Scripts and results table

Heatmap 5A using the package `{ComplexHeatmap}`

```{r}
mrlog_rorder <- vroom::vroom_lines("PTK2_nachbarn.txt")
# takes time, compute the regularized logs.
rld_mat <- rlog(dds, blind = FALSE)
# annotate with metadata
as_tibble(colData(dds), rownames = "id") %>% 
  # cleanup names and get them shorter
  mutate(cond = str_replace(condition, "-", "_") %>%
           str_replace("50", "") %>% 
           str_replace("A375", "A") %>% 
           str_replace("WM1346", "W")) %>% 
  group_by(cond) %>% 
  mutate(rep = row_number()) %>% 
  unite(c(cond, rep), col = "samples") -> col_data
rlog_mat <- assay(rld_mat)
colnames(rlog_mat) <- col_data$samples
# extract only the 63 genes that are neighbors of PTK2
# first get the gene_ids from the gene symbols
filter(IZI, gene_name %in% mrlog_rorder) %>% 
  distinct(gene_id) %>% 
  pull(gene_id) -> nachbarn_ids

# second extract using the gene ids
rlog_mat[rownames(rlog_mat) %in% nachbarn_ids, ] %>% 
  as_tibble(rownames = "gene_id") %>% 
  # annotate back with gene_name
  left_join(filter(res_tib_WM1346_izi, gene_name %in% mrlog_rorder) %>% 
              select(starts_with("gene")) %>% 
              distinct()) %>% 
  # tidy format
  pivot_longer(cols = -starts_with("gene"), 
               names_to = c("condition", "rep"),
               names_pattern = "(.+)_([123])$",
               values_to = "rlog") %>% 
  group_by(across(gene_id:condition)) %>% 
  # compute mean/sd for the 3 replicates per condition
  summarise(mrlog = mean(rlog),
            sd_rlog = sd(rlog), .groups = "drop") %>%
  select(-gene_id) %>% 
  # for heatmap, we need a wide format
  pivot_wider(id_cols = gene_name,
              names_from = condition,
              values_from = c(mrlog, sd_rlog),
              names_glue = "{condition}_{.value}") %>%
  select(gene_name, pA_mrlog, pA_sd_rlog, cA_mrlog, cA_sd_rlog,
                    pA_IZI_mrlog, pA_IZI_sd_rlog, cA_IZI_mrlog, cA_IZI_sd_rlog, 
                    pW_mrlog, pW_sd_rlog, cW_mrlog, cW_sd_rlog,
             pW_IZI_mrlog, pW_IZI_sd_rlog, cW_IZI_mrlog, cW_IZI_sd_rlog) -> m_sd 
m_sd %>% 
  # to get consistent row ordering, fix it here
  dplyr::slice(order(factor(gene_name, levels = mrlog_rorder))) %>% 
  select(-contains("sd")) %>% 
  rename_with(~ str_replace(.x, "_mrlog", "")) %>% 
  column_to_rownames(var = "gene_name") %>% 
  as.matrix() -> nachbarn_mat
# same for column order, disable clustering
my_col_order <- c("pA",  "pA_IZI", "cA", "cA_IZI",
                    "pW", "pW_IZI", "cW", "cW_IZI")
Heatmap(nachbarn_mat,
        col = circlize::colorRamp2(c(min(nachbarn_mat), mean(nachbarn_mat), max(nachbarn_mat)),
                           c("blue", "white", "red")),
        show_column_dend = FALSE,
        column_order = my_col_order,
        column_names_gp = gpar(fontsize = 14),
        column_names_centered = TRUE,
        column_names_rot = 0,
        name = "rlog") -> ht_nachbarn_mat
ComplexHeatmap::draw(ht_nachbarn_mat)
```


The object `m_sd` was used to plot with GraphPad Prism the figure 4C, 5B, 6D.


### GeneWalk


Fig 4B, plotting code:

```{r, fig.width=10, fig.height=6}
vroom::vroom("genewalk_scatterplots.csv",
             col_types = cols()) %>% 
  ggplot(aes(x = go_con, y = frac_rel_go, size = gene_con, colour = frac_rel_go)) +
  geom_point() +
  scale_x_log10() +
  scale_size_area() +
  scale_color_viridis_c(option = "C") +
  geom_text(data = function(x) filter(x, gene_con > 30, go_con > 30),
            aes(label = hgnc_symbol), colour = "black", size = 4, nudge_x = .08) +
  geom_hline(yintercept = 0.5, linetype = "dashed", colour = "grey50") +
  geom_vline(xintercept = 30, linetype = "dashed", colour = "grey50") +
  annotation_logticks(sides = "b") +
  theme_light(16, base_family = "RobotoCondensed") +
  labs(x = "Number of GO annotations (per gene)",
       y = "Fraction of relevant GO terms (per gene)",
       title = "Moonlighting genes",
       size = "Gene connections",
       colour = "Fraction of relevant GO terms (per gene)") +
  theme(legend.position = "top") +
  guides(color = guide_colorbar(title.position = "top", title.hjust = .5,
                                barwidth = unit(20, "lines"), barheight = unit(.5, "lines")),
         size = guide_legend(title.position = "top", title.hjust = .5)) -> moon
moon
```






